lib/fluent/plugin/filter_masking.rb

```diff
@@ -15,11 +15,18 @@ def strToHash(str)
       # error safe method - if any error occurs the original record is returned
       def maskRecord(record)
         maskedRecord = record
-        
-        begin
+        excludedFields = []
+        @fieldsToExcludeJSONPathsArray.each do | field |
```

Comment:
`field.split(".")` can be done in the `configure` stage and that way it will be called only once and not on every record that comes in which will improve performance

Reply:
changed